pkg/server/grpc/grpc.go

```diff
@@ -127,3 +127,16 @@ func (s *Server) Shutdown(err error) {
 		cancel()
 	}
 }
+
+// ReadWriteStoreServer is a StoreServer and a WriteableStoreServer.
+type ReadWriteStoreServer interface {
+	storepb.StoreServer
+	storepb.WriteableStoreServer
```

Comment:
I am wondering again about exposing these on the same gRPC service. If we do, we have to make sure to document that exposing the receive gRPC to a WAN to enable writes means that one tenant could read any other tenant’s metrics

Reply:
I guess that's acceptable for now 🤔